scripts/install-nix-from-closure.sh

```diff
@@ -188,6 +188,15 @@ fi
 # shellcheck source=./nix-profile.sh.in
 . "$nix/etc/profile.d/nix.sh"
 
+NIX_LINK="$HOME/.nix-profile"
+if ! [ -e "$NIX_LINK" ]; then
+    if [ -n "$XDG_STATE_HOME" ]; then
+        NIX_LINK="$XDG_STATE_HOME/nix/profile"
+    else
+        NIX_LINK="$HOME/.local/state/nix/profile"
+    fi
+fi
+
```

Comment:
I think that should go after the `nix-env -i "$nix"` call just after since (afaik) Nix won't have a chance to create that directory in the first place, so it'd alway default to the XDG version (which we don't want)